### PR TITLE
RJAVA-564: Use monorepo

### PR DESCRIPTION
**Pull request by fealebenpae**

### CODE REVIEW

**Comment by fealebenpae** [comment]
@cmelchior @clementetb I think I've done everything I can. Please let me know if there is anything else I can help with before you start building using the monorepo.

N.B. this sets up Core as a submodule and builds it from source. I will submit another PR that implements downloading prebuilt Core artifacts once we start releasing those.

**Comment by clementetb** [comment]
@fealebenpae 

Yes, there is something else. Realm-java validation fails on JSON formatting tests because ObjectId elements miss of an ending quotation mark:

https://github.com/realm/realm-core/blob/cf38093f5fcc7252c6305afaa74997bdb981754d/src/realm/obj.cpp#L817

Could you patch it?

File: realm/realm-library/src/main/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -131,32 +130,19 @@ create_javah(TARGET jni_headers
...
-include(${PROJECT_BINARY_DIR}/lib/cmake/OpenSSL/OpenSSLConfig.cmake)
-get_target_property(openssl_include_DIR OpenSSL::Crypto INTERFACE_INCLUDE_DIRECTORIES)
-get_target_property(crypto_LIB OpenSSL::Crypto IMPORTED_LOCATION)
-get_target_property(ssl_LIB OpenSSL::SSL IMPORTED_LOCATION)
+option(REALM_JAVA_BUILD_CORE_FROM_SOURCE "Build Realm Core from source, as opposed to downloading prebuilt binaries" ON)
+if(REALM_JAVA_BUILD_CORE_FROM_SOURCE)
+    set(REALM_BUILD_LIB_ONLY ON)
+    add_subdirectory(realm-core)
+else()
+    message(FATAL_ERROR "TODO: Implement downloading monorepo release artifacts")
```
Comment (cmelchior) [inline realm/realm-library/src/main/cpp/CMakeLists.txt:138]: @fealebenpae Any idea about a timeline when these are available?

**Review by cmelchior — APPROVED** [review]
Really nice 🎉 . Particularly that building against source works from the start, although I am a bit worried about what that will mean for our CI build speed 🙈, so hopefully, we can have prebuilt binaries soon.

File: realm/realm-library/src/main/cpp/CMakeLists.txt (hunk 1 above)
Comment (fealebenpae) [reply realm/realm-library/src/main/cpp/CMakeLists.txt:138]: I want to iron out all the kinks in the integration before I cut binary releases, so as soon as I've fixed the other issues (like the Mixed serialization above).

**Comment by cmelchior** [comment]
Looks like we are not respecting the `-PbuildTargetABIs=x86_64` flags when building from monorepo source. Will take a look and see if it is easy to add as it has a very big impact on local build speeds.

**Comment by beeender** [comment]
:tada: :tada: :tada:

**Pull request merged**